Modules/_functoolsmodule.c

```diff
@@ -41,13 +42,39 @@ get_functools_state(PyObject *module)
 
 /* partial object **********************************************************/
 
+
+typedef struct {
+    PyObject_HEAD
+} placeholderobject;
+
+
+PyDoc_STRVAR(placeholder_doc, "placeholder for partial arguments");
+
+
+static PyType_Slot placeholder_type_slots[] = {
+    {Py_tp_doc, (void *)placeholder_doc},
+    {0, 0}
+};
+
+static PyType_Spec placeholder_type_spec = {
+    .name = "partial.Placeholder",
+    .basicsize = sizeof(placeholderobject),
+    .flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE |
```

Comment:
```suggestion
    .flags = Py_TPFLAGS_DEFAULT |
```

This also needs to be tested. See the `test.support.check_disallow_instantiation` helper.